src/container.cpp

```diff
@@ -446,63 +444,63 @@ Cylinder* Container::queryDestination(int32_t& index, const Thing& thing, Item**
 		the slot position as if the bag has 20 slots
 		*/
 		index = INDEX_WHEREEVER;
-		*destItem = nullptr;
+		destItem = nullptr;
 	}
 
-	const Item* item = thing.getItem();
+	auto item = thing->getItem();
 	if (!item) {
-		return this;
+		return shared_from_this();
 	}
 
 	if (index != INDEX_WHEREEVER) {
-		Item* itemFromIndex = getItemByIndex(index);
+		auto itemFromIndex = getItemByIndex(index);
 		if (itemFromIndex) {
-			*destItem = itemFromIndex;
+			destItem = itemFromIndex;
 		}
 
-		Cylinder* subCylinder = dynamic_cast<Cylinder*>(*destItem);
-		if (subCylinder) {
+		if (destItem) {
+			auto subCylinder = destItem;
```

Comment:
Could this be `const auto`
```suggestion
			const auto subCylinder = destItem;
```